09_Hyper-parameter_tuning_using_Ray/reference_solution/GPT-neo-ray-tune.py

```diff
@@ -173,12 +173,28 @@ def tokenize(x):
         "args": args,
     }
 
-    analysis = tune.run(
-        model_training,
-        resources_per_trial={"cpu": 7, "gpu": 1},  # set resources for every trial run
-        config=config,
-        num_samples=8,
-        metric="perplexity",
-        mode="min",
+    # Create a Tuner object
+    tuner = tune.Tuner(
+        tune.with_resources(
+            model_training,
+            # Note, to code reviewer: the accelerator_type specification
+            # has no influence on the performance but shows 
+            # 1.0/1.0 accelerator_type:AMD-Instinct-MI250X in the output file.
+            # Without the specification we get 0.0/1.0 accelerator_type:AMD-Instinct-MI250X
+            # I am conflicted whether we should keep or remove this.
+            resources={"cpu": 7, "gpu": 1, "accelerator_type:AMD-Instinct-MI250X": 0.125},  # Set resources for every trial run
```

Comment:
We should probably just remove it and ignore this output for now. Looks a bit like a bug to me - AMD support in Ray is still considered experimental (https://docs.ray.io/en/latest/ray-core/scheduling/accelerators.html).